back/engines/commercial/user_custom_fields/spec/acceptance/ref_distributions_spec.rb

```diff
@@ -0,0 +1,159 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'rspec_api_documentation/dsl'
+
+resource 'Representativeness reference distributions' do
+  header 'Content-Type', 'application/json'
+
+  shared_examples 'unauthorized requests' do
+    context 'when visitor' do
+      example 'returns 401 (Unauthorized)', document: false do
+        do_request
+        expect(status).to eq(401)
+      end
+    end
+
+    context 'when normal user' do
+      before { user_header_token }
+
+      example 'returns 401 (Unauthorized)', document: false do
+        do_request
+        expect(status).to eq(401)
+      end
+    end
+  end
+
+  get 'web_api/v1/users/custom_fields/:custom_field_id/reference_distribution' do
+    let!(:reference_distribution) { create(:ref_distribution) }
+    let(:custom_field_id) { reference_distribution.custom_field.id }
+
+    context 'when admin' do
+      before { admin_header_token }
+
+      context 'when the custom field has a reference distribution' do
+        let(:expected_response) do
+          {
+            data: {
+              id: reference_distribution.id,
+              type: 'reference_distribution',
+              attributes: {
+                distribution: reference_distribution.probabilities_and_counts.symbolize_keys
+              },
+              relationships: {
+                values: {
+                  data: reference_distribution.distribution.keys.map do |key|
+                    { type: 'custom_field_option', id: key }
+                  end
+                }
+              }
+            }
+          }
+        end
+
+        example_request 'Get the reference distribution associated to a custom field' do
+          expect(status).to eq(200)
+          expect(json_response_body).to match(expected_response)
+        end
+      end
+
+      context 'when the custom field does not have a reference distribution' do
+        before { reference_distribution.destroy! }
+
+        example_request 'returns 404 (Not Found)' do
+          expect(status).to eq(404)
+        end
+      end
+    end
+
+    include_examples 'unauthorized requests'
+  end
+
+  post 'web_api/v1/users/custom_fields/:custom_field_id/reference_distribution' do
+    parameter :distribution, 'The distribution as (unnormalized) counts', required: true
+
+    let(:custom_field_id) { custom_field.id }
+    let(:custom_field) do
+      create(:custom_field_select, resource_type: 'User').tap do |cf|
```

Comment:
Please use at least 3 letters for a variable name. Soon 2 letters will not pass RuboCop. `custom_field` is even better, because the reader does not have to think about what the variable represents.

Reply:
sad 😢

Reply:
Why does it make you sad?

Reply:
Code is more often read than written, so "even helps to make the code easier to read" is the main goal of using good names. The fact we have longer lines is a side effect, but for the better.

Reply:
Allow me to disagree that all short names are bad names.